osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -99,6 +102,28 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
         /// </summary>
         private readonly Queue<Action> postQueue = new Queue<Action>();
 
+        /// <summary>
+        /// Scans for inline commands of the form $cmd and returns the newly processed text
+        /// </summary>
+        public string ProcessInlineCommands(string text) {
+
+            return Regex.Replace(text, @"\$[a-zA-Z0-9]+", delegate(Match match) {
```

Comment:
this is very inefficient and does not need to be run on lines that don't contain a `$`.

Reply:
Any idea what should I do to improve the efficiency of this ? I wanted to add this regex in MessageFormatter but I cannot inject other components since it's a static class.

Reply:
a good start would be avoiding the regex if there's no "$" in the string, as i mentioned.